lib/wallet/integration/src/Test/Integration/Scenario/API/Shared/Transactions.hs

```diff
@@ -1730,6 +1740,237 @@ spec = describe "SHARED_TRANSACTIONS" $ do
+                , expectField #collateral (`shouldBe` [])
+                , expectField #scriptValidity
+                    (`shouldBe` (Just $ ApiT TxScriptValid))
+                , expectField #depositsReturned (`shouldBe` [])
+                , expectField #depositsTaken (`shouldBe` [depositAmt])
+                ]
+        let (ApiScriptTemplate pScriptTemplate) =
+                party1 ^. #paymentScriptTemplate
+        let paymentScript =
+                ApiT $ NativeExplicitScript
+                (replaceCosignersWithVerKeys
+                    CA.UTxOExternal pScriptTemplate (Index 1))
+                ViaSpending
+        let (Just (ApiScriptTemplate dScriptTemplate)) =
+                party1 ^. #delegationScriptTemplate
+        let delegationScript =
+                ApiT $ NativeExplicitScript
+                (replaceCosignersWithVerKeys
+                    CA.Stake dScriptTemplate (Index 0))
+                ViaSpending
+        let witsExp1 =
+                [ expectField (#witnessCount . #scripts)
+                      (`shouldContain` [delegationScript])
+                , expectField (#witnessCount . #scripts)
+                      (`shouldContain` [paymentScript])
+                , expectField (#witnessCount . #verificationKey)
+                      (`shouldBe` 0)
+                , expectField (#witnessCount . #bootstrap)
+                      (`shouldBe` 0)]
+
+        let certExpectation1 =
+                [expectField #certificates
+                     (`shouldBe` [ registerStakeKeyCert stakeKeyDerPathParty1
+                                 , delegatingCert stakeKeyDerPathParty1])]
+        verify rDecodedTx1 (decodedExpectations ++ certExpectation1 ++ witsExp1)
+        let certExpectation2 =
+                [expectField #certificates
+                     (`shouldBe` [ registerStakeKeyCert stakeKeyDerPathParty2
+                                 , delegatingCert stakeKeyDerPathParty2])]
+        verify rDecodedTx2 (decodedExpectations ++ certExpectation2 ++ witsExp1)
+        let (ApiSerialisedTransaction apiTx1 _) =
+                getFromResponse #transaction rTx1
+        signedTx1 <-
+            signSharedTx ctx party1 apiTx1
+                [ expectResponseCode HTTP.status202 ]
+        let decodePayload2 = Json (toJSON signedTx1)
+        rDecodedTx3 <- request @(ApiDecodedTransaction n) ctx
+            (Link.decodeTransaction @'Shared party1) Default decodePayload2
+        rDecodedTx4 <- request @(ApiDecodedTransaction n) ctx
+            (Link.decodeTransaction @'Shared party2) Default decodePayload2
+
+        -- there is now 1 witness for staking and 1 witness for payment from party1
+        let witsExp2 =
+                [ expectField (#witnessCount . #scripts)
+                      (`shouldContain` [delegationScript])
+                , expectField (#witnessCount . #scripts)
+                      (`shouldContain` [paymentScript])
```

Comment:
Neat! 

Perhaps we could also make sure there aren't any more scripts there through something like this? (haven't tested it)
```suggestion
                , expectField (#witnessCount . #scripts . Set.fromList)
                      (`shouldBe` (Set.fromList [paymentScript, delegationScript]))
```